### PR TITLE
[Filter/Custom-Easy] Custom-Easy Implementation

### PR DESCRIPTION
**Pull request by myungjoo**
Developers may insert a custom filter with a single C function.
In the future, with C++ wrapper, we can insert a LAMBDA function
to an NNStreamer pipeline with a single-line of code.

Ultimate Objective:
```
NNS_custom_easy_register_cpp ("doit", lambda(x, data->f(x, data), where x is a tensor), data, 100:100, 100:100);
gst-parse-launch (" ... ! tensor_filter framework=custom-easy model=doit ! ..."
```

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

### CODE REVIEW

**Comment by myungjoo** [comment]
Test Cases and Initial Fixes will Follow.

**Comment by myungjoo** [comment]
> Test Cases and Initial Fixes will Follow.

Done.

File: gst/nnstreamer/tensor_filter/tensor_filter_common.c [hunk 1]
```diff
@@ -423,7 +423,10 @@ gst_tensor_filter_common_set_property (GstTensorFilterPrivate * priv,
       model_num = gst_tensor_filter_parse_modelpaths_string (prop, model_files);
       if (model_num == 1) {
         if (!g_file_test (prop->model_file, G_FILE_TEST_IS_REGULAR))
-          g_critical ("Cannot find the model file: %s\n", prop->model_file);
+          g_critical ("Cannot find the model file: %s\n"
+              "Ignore this if you have specified name and the model is preloaded.\n"
+              "Note that this should be refactored.\n", prop->model_file);
+          /** @todo This mechanism should be refactored. */
```
Comment (kparichay) [inline gst/nnstreamer/tensor_filter/tensor_filter_common.c]: is this the case for the frameworks which have set `run_without_model` as TRUE?

Comment (myungjoo) [reply gst/nnstreamer/tensor_filter/tensor_filter_common.c]: Nope.

It's just not a path, but a name.

File: gst/nnstreamer/tensor_filter/tensor_filter_custom_easy.c [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  internal_data *ptr;
+
+  ptr = g_new0 (internal_data, 1);
+
+  if (!ptr)
+    return -ENOMEM;
+
+  ptr->func = func;
+  ptr->data = data;
+  ptr->in_info = in_info;
```
Comment (kparichay) [inline gst/nnstreamer/tensor_filter/tensor_filter_custom_easy.c]: Should there be sanity checks for these variables (`in_info`, `out_info` and `func`), as in later functions we have asserts verifying that they exist?

Comment (myungjoo) [reply gst/nnstreamer/tensor_filter/tensor_filter_custom_easy.c]: Yes they are needed.

Comment (myungjoo) [reply gst/nnstreamer/tensor_filter/tensor_filter_custom_easy.c]: and fixed :)

**Pull request merged**